feather/commands/src/impls.rs

```diff
@@ -0,0 +1,868 @@
+//     //        .remove(&uuid.to_hyphenated_ref().to_string());
+//     //}
+//
+//     if let Ok(mut chat) = ctx.ecs.get_mut::<ChatBox>(ctx.sender) {
+//         let kick_confirm =
+//             Text::translate_with("commands.pardon.success", vec![Text::from(name.0)]);
+//         chat.send_system(kick_confirm);
+//     }
+//
+//     Ok(None)
+// }
+//
+// #[derive(Debug, Error)]
+// pub enum PardonIpError {
+//     #[error("Invalid IP Address")]
+//     NotIp,
+// }
+//
+// #[command(usage = "pardon-ip <ip>")]
+// pub fn pardonip(ctx: &mut CommandCtx, ip: String) -> anyhow::Result<Option<String>> {
+//     // Try to parse ip
+//     let _addr = IpAddr::from_str(&ip).map_err(|_| PardonIpError::NotIp)?;
+//
+//     //{
+//     //    let mut bi_lock = ctx.game.resources.get_mut::<WrappedBanInfo>();
+//     //    let mut ban_info = bi_lock.write().unwrap();
+//     //    ban_info.ip_bans.remove(&addr);
+//     //}
+//
+//     if let Ok(mut chat) = ctx.ecs.get_mut::<ChatBox>(ctx.sender) {
+//         let kick_confirm = Text::translate_with("commands.pardon.success", vec![Text::from(ip)]);
+//         chat.send_system(kick_confirm);
+//     }
+//
+//     Ok(None)
+// }
+//
+// #[command(usage = "time query <info>")]
+// pub fn time_query(
+//     ctx: &mut CommandCtx,
+//     info: TimeQueryInformation,
+// ) -> anyhow::Result<Option<String>> {
+//     let time = match info {
+//         TimeQueryInformation::DayTime => ctx.world.time.time(),
+//         TimeQueryInformation::GameTime => ctx.world.time.world_age(),
+//         TimeQueryInformation::Day => ctx.world.time.days(),
+//     };
+//
+//     if let Ok(mut chat) = ctx.ecs.get_mut::<ChatBox>(ctx.sender) {
+//         let message =
+//             Text::translate_with("commands.time.query", vec![Text::from(time.to_string())]);
+//         chat.send_system(message);
+//     }
+//
+//     Ok(None)
+// }
+//
+// #[command(usage = "time add <time>")]
+// pub fn time_add(ctx: &mut CommandCtx, time: TimeArgument) -> anyhow::Result<Option<String>> {
+//     let time = ctx.world.time.time() + time.0;
+//     set_time(ctx, time);
+//     Ok(None)
+// }
+//
+// #[command(usage = "time set <time>")]
+// pub fn time_set_0(ctx: &mut CommandCtx, time: TimeArgument) -> anyhow::Result<Option<String>> {
+//     set_time(ctx, time.0);
+//     Ok(None)
+// }
+//
+// #[command(usage = "time set <time_spec>")]
+// pub fn time_set_1(ctx: &mut CommandCtx, time_spec: TimeSpec) -> anyhow::Result<Option<String>> {
+//     set_time(
+//         ctx,
+//         match time_spec {
+//             TimeSpec::Day => 1_000,
+//             TimeSpec::Noon => 6_000,
+//             TimeSpec::Night => 13_000,
+//             TimeSpec::Midnight => 18_000,
+//         },
+//     );
+//     Ok(None)
+// }
+//
+// fn set_time(ctx: &mut CommandCtx, time: u64) {
+//     ctx.ecs.insert_event(TimeUpdateEvent {
+//         old: ctx.world.time.time(),
+//         new: time,
+//     });
+//     ctx.world.time.set_time(time);
+// }
```

Comment:
We might want to look into a different way to organize commands instead of a monolithic implementation file.